shell/platform/android/flutter_main.cc

```diff
@@ -211,4 +229,53 @@ bool FlutterMain::Register(JNIEnv* env) {
   return env->RegisterNatives(clazz, methods, fml::size(methods)) == 0;
 }
 
+// static
+AndroidRenderingAPI FlutterMain::SelectedRenderingAPI(
+    const flutter::Settings& settings) {
+  if (settings.enable_software_rendering) {
+    FML_CHECK(!settings.enable_impeller)
+        << "Impeller does not support software rendering. Either disable "
+           "software rendering or disable impeller.";
+    return AndroidRenderingAPI::kSoftware;
+  }
+  // Debug/Profile only functionality for testing a specific
+  // backend configuration.
+#ifndef FLUTTER_RELEASE
+  if (settings.requested_rendering_backend == "opengles" &
+      settings.enable_impeller) {
+    return AndroidRenderingAPI::kImpellerOpenGLES;
+  }
+  if (settings.requested_rendering_backend == "vulkan" &&
+      settings.enable_impeller) {
+    return AndroidRenderingAPI::kImpellerVulkan;
+  }
+#endif
+
+  if (settings.enable_impeller) {
+    // Vulkan must only be used on API level 29+, as older API levels do not
+    // have requisite features to support platform views.
+    //
+    // Even if this check returns true, Impeller may determine it cannot use
+    // Vulkan for some other reason, such as a missing required extension or
+    // feature.
+    int api_level = android_get_device_api_level();
+    if (api_level < kMinimumAndroidApiLevelForVulkan) {
+      return AndroidRenderingAPI::kImpellerOpenGLES;
+    }
+    // Determine if Vulkan is supported by creating a Vulkan context and
+    // checking if it is valid.
+    impeller::ScopedValidationDisable disable_validation;
+    auto vulkan_backend = std::make_unique<AndroidContextVulkanImpeller>(
+        /*enable_vulkan_validation=*/false,
+        /*enable_vulkan_gpu_tracing=*/false,
+        /*quiet=*/true);
+    if (!vulkan_backend->IsValid()) {
+      return AndroidRenderingAPI::kImpellerOpenGLES;
```

Comment:
We can implement ImpellerVulkan -> Skia fallback by changing this value (or adding another configuration setting).